### ISSUE TITLE
rank_features() for supervised Splitset

### ISSUE THREAD

**Issue by aiqc** [aiqc/aiqc#37]
> Difficulty: ★★☆☆☆

## Background
If `Splitset.supervision=='supervised'` then it is guaranteed to have both a `Label` and a `Featureset`.

## Problem
There is currently no way to make informed decisions about which features should be carried forward into the neural network training. This is not only wastes time/compute, but also leads to more complex models and worse performance. 

## Solution
- New method on the `Splitset` class called `rank_features()`. 
- It would take the Label and the Featureset and run.
  - [sklearn.inspection.permuted_feature_importance()](https://scikit-learn.org/stable/modules/permutation_importance.html#permutation-feature-importance)

## Discussion
- Determining the type of analysis based on Label could get tricky. Based on the Label's dtype you could determine if it was intended for regression, classification binary, or classification multi, but we already have this logic in `Batch.analysis_type`
- We should move `analysis_type` attribute out of `Batch` class and into `Splitset` class to support this.

**Comment by aiqc** [comment]
Check out the [`datum` module](https://aiqc.readthedocs.io/en/latest/notebooks/example_datasets.html) for built-in datasets.

And the [`tests` module](https://github.com/aiqc/aiqc/blob/main/.github/pull_request_template.md) (described in PR process) can help you make Splitsets for different types of analysis instantly.
